src/mbgl/text/cross_tile_symbol_index.cpp

```diff
@@ -169,6 +169,18 @@ bool CrossTileSymbolIndex::addLayer(RenderSymbolLayer& symbolLayer) {
     return symbolBucketsChanged;
 }
 
+void CrossTileSymbolIndex::pruneUnusedLayers(const std::set<std::string>& usedLayers) {
+    std::vector<std::string> unusedLayers;
+    for (auto layerIndex : layerIndexes) {
+        if (usedLayers.find(layerIndex.first) == usedLayers.end()) {
+            unusedLayers.push_back(layerIndex.first);
+        }
+    }
+    for (auto unusedLayer : unusedLayers) {
+        layerIndexes.erase(unusedLayer);
+    }
+}
+
```

Comment:
I think this could be done in a single loop like in [`removeStaleBuckets`](https://github.com/mapbox/mapbox-gl-native/blob/f06eb0a6cfc1ff746959ce864aa5a35a5e080c92/src/mbgl/text/cross_tile_symbol_index.cpp#L125-L132).

On second thought, this approach might be simpler here. Maybe we should change `removeStaleBuckets`.

Reply:
Yeah, I'm not sure, it feels like it should be one loop, but doing it the `removeStaleBuckets` way has so much iterator management...